### PR TITLE
Redesign plot extreme + some minor changes

### PR DESCRIPTION
**Pull request by algmyr**

### CODE REVIEW

File: tle/cogs/graphs.py [hunk 1]
```diff
@@ -96,12 +96,17 @@ def _filter_solved_submissions(submissions, contests, tags = []):
 def get_extremes(user, submissions, problems):
```
Comment (krofna) [inline tle/cogs/graphs.py]: dead argument

File: tle/cogs/graphs.py [hunk 2]
```diff
@@ -93,15 +93,20 @@ def _filter_solved_submissions(submissions, contests, tags = []):
                 problems.add(problem_key)
     return solved_subs
 
-def get_extremes(user, submissions, problems):
+def get_extremes(submissions, problems):
 
     def check(sub):
-        return sub.verdict == 'OK' and sub.author.participantType == 'CONTESTANT' and sub.problem.rating is not None
+        return sub.verdict == 'OK' and \
```
Comment (meooow25) [inline tle/cogs/graphs.py]: Parentheses are preferred over backslashes by PEP 8
```py
return (sub.verdict == 'OK' and
        sub.author.participantType == 'CONTESTANT' and
        sub.problem.rating is not None)
```

File: tle/cogs/graphs.py [hunk 3]
```diff
@@ -132,41 +136,72 @@ def _running_mean(x, bin_size):
...
+    times = [
+        dt.datetime.fromtimestamp(rating_change.ratingUpdateTimeSeconds)
+        for rating_change in rating_changes
+    ]
+
+    regular = []
+    fullsolves = []
+    nosolves = []
+    for t,extreme in zip(times, extremes):
+        mn,mx = extreme
```
Comment (meooow25) [inline tle/cogs/graphs.py]: Minor style issue: add spaces after commas

File: tle/cogs/graphs.py [hunk 4]
```diff
@@ -269,13 +304,13 @@ def __init__(self, bot):
                 statuses.append([])
             statuses[-1].append(submission)
 
-        problemsets = [problems for problems in [cf_common.cache2.contest_cache.get_problems(contest_id=c.id) for c in contests]]  
+        problemsets = [problems for problems in [cf_common.cache2.contest_cache.get_problems(contest_id=c.id) for c in contests]]
 
         plt.clf()
-        _plot_extreme(await cf.user.info(handles=set(handles)), resp, statuses, problemsets)
+        users = await cf.user.info(handles=set(handles))
```
Comment (meooow25) [inline tle/cogs/graphs.py]: `set(handles)` is unnecessary, handles has length 1 always.

File: tle/cogs/graphs.py (hunk 3 above)
Comment (algmyr) [reply tle/cogs/graphs.py]: There is a lot of cleanup to do if we want to enforce that everywhere (although in this file most of it is probably due to me...). I fixed it in the parts relevant to this PR.

File: tle/cogs/graphs.py (hunk 4 above)
Comment (algmyr) [reply tle/cogs/graphs.py]: In my defense, not my code. Arguably I should have cleaned it up though.

File: tle/cogs/graphs.py (hunk 2 above)
Comment (algmyr) [reply tle/cogs/graphs.py]: Done.

**Pull request merged**